src/ArcxAnalyticsSdk.ts

```diff
@@ -75,6 +95,80 @@ export class ArcxAnalyticsSdk {
     )
   }
 
+  private async _onAccountsChanged(accounts: string[]) {
+    if (!window.ethereum) {
+      throw new Error('ArcxAnalyticsSdk::_onAccountsChanged: No ethereum provider found')
+    }
+
+    if (accounts.length > 0) {
+      if (accounts[0] === this.previousConnectedAccount) {
+        // We have already reported this account
+        return
+      } else {
+        this.previousConnectedAccount = accounts[0]
+      }
+
+      const chainIdHex = await window.ethereum.request<string>({ method: 'eth_chainId' })
```

Comment:
why not `window.ethereum.chainId`

Reply:
It's deprecated: https://docs.metamask.io/guide/ethereum-provider.html#ethereum-chainid-deprecated

Reply:
should we put chainId here anyway? Shouldn't `CHAIN_CHANGED` catch it?